### ISSUE TITLE
Real time HP bar 

### ISSUE THREAD

**Issue by nlruk** [runelite/runelite#11413]
I noticed that when I eat food the status bar accounts for it but my hp bar above my head doesn't straight away.

Can you please remove the delay so the hp bar reflects what the status bar is saying? The real value.

You probably won't be affected by this issue unless you have very low HP. I'm a skiller with 10 HP so the delay is substantial when trying to stay alive

<img width="538" alt="Screenshot 2020-04-30 at 07 19 29" src="https://user-images.githubusercontent.com/63312904/80679124-2eac1f80-8ab4-11ea-909e-71c7b8f2cc6b.png">

**Comment by Nightfirecat** [comment]
This would mean that your own HP bar would behave differently from _every other HP bar in the game._ While this is technically possible, I'm against this.

**Comment by nlruk** [comment]
> This would mean that your own HP bar would behave differently from _every other HP bar in the game._ While this is technically possible, I'm against this.

Why would that matter? It's optional and it is very beneficial to people who actually play the game.

**Comment by Nightfirecat** [comment]
It matters because it's not a good use of our time to add every possible feature anyone could want. I don't see this as a feature that "people who actually play the game" would universally want. For instance, I play the game plenty and would not want this.

**Comment by abextm** [comment]
This is a bad idea because it would be inaccurate and confusing relative to every other healthbar.

**Comment by nlruk** [comment]
> This is a bad idea because it would be inaccurate and confusing relative to every other healthbar.

It's not inaccurate though. It would be displaying the true value.

**Comment by nlruk** [comment]
> This is a bad idea because it would be inaccurate and confusing relative to every other healthbar.

I understand your perspective from a programming/time allocation point of view but if you played the game actively you would know you're incorrect.

It's not inaccurate. It would be displaying the real value as I said above there is a delay which is very noticeable if you have low HP.

**Comment by Matsyir** [comment]
> if you played the game actively you would know you're incorrect.

No need for the condescending tone.. I don't know a single RL dev who _doesn't_ play the game, not sure why you'd assume that's the case.

> It's not inaccurate. It would be displaying the real value

Correct, but it _is_ inaccurate when compared to every other health bar except yours, which you've probably gotten used to their behaviour over the years, even if you don't realize it. This would make for some very odd health bars while PKing, you'd assume your opponent is super slow at eating or something, you'd think you can combo him out all the time, with their health bar always seeming low since their eats don't update right away, while yours do. Same with group bossing, you'd be like wtf why is everyone risking it? Unless you constantly kept this in mind, ofc - but it's something extra to think about when you likely already have enough on your plate. At 10hp, shouldn't you just be eating nearly any time you get hit >0 anyways, in the majority of cases where it actually matters? You can also just use the status bar, and you can also get notifications once you go under a certain HP from the Idle Notifier plugin.

In a nutshell, I agree with abex, and I'm fairly sure you would too if you tried something like this. I don't think you realize how 'off' it would feel having your health bar behave completely differently than others'.

Personally, I'm all about "why not" when it comes to external plugins - if it's not malicious and it works as intended - sure why not. But I really don't think this would be as useful as you imagine. I won't be working on it and I can't approve external plugins regardless, but that's my perspective.

**Comment by nlruk** [comment]
> > if you played the game actively you would know you're incorrect.
> 
> No need for the condescending tone.. I don't know a single RL dev who _doesn't_ play the game, not sure why you'd assume that's the case.
> 
> > It's not inaccurate. It would be displaying the real value
> 
> Correct, but it _is_ inaccurate when compared to every other health bar except yours, which you've probably gotten used to their behaviour over the years, even if you don't realize it. This would make for some very odd health bars while PKing, you'd assume your opponent is super slow at eating or something, you'd think you can combo him out all the time, with their health bar always seeming low since their eats don't update right away, while yours do. Same with group bossing, you'd be like wtf why is everyone risking it? Unless you constantly kept this in mind, ofc - but it's something extra to think about when you likely already have enough on your plate. At 10hp, shouldn't you just be eating nearly any time you get hit >0 anyways, in the majority of cases where it actually matters? You can also just use the status bar, and you can also get notifications once you go under a certain HP from the Idle Notifier plugin.
> 
> In a nutshell, I agree with abex, and I'm fairly sure you would too if you tried something like this. I don't think you realize how 'off' it would feel having your health bar behave completely differently than others'.
> 
> Personally, I'm all about "why not" when it comes to external plugins - if it's not malicious and it works as intended - sure why not. But I really don't think this would be as useful as you imagine. I won't be working on it and I can't approve external plugins regardless, but that's my perspective.

Fair enough and I understand. I just felt my ideas are automatically disregarded especially when the same people keep thumbs downing it within seconds.
